spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/GeoConverters.java

```diff
@@ -117,6 +117,10 @@ public Point convert(DBObject source) {
 
 			Assert.isTrue(source.keySet().size() == 2, "Source must contain 2 elements");
 
+			if (source.containsField("type")) {
```

Comment:
This change seems to be unrelated to the PR. @christophstrobl — can you comment?